contracts/Scales.sol

```diff
@@ -238,6 +239,9 @@ contract Scales is IScales, ERC20, ERC721Holder, AccessControl, Pausable, Reentr
         uint16 genesisCount;
         for (uint256 i; i < tokenIds.length; i++) {
             uint256 tokenId = tokenIds[i];
+            
+            // Check msg.sender is owner of tokenId staked in contract
+            if ( tokenOwners[tokenId] != _msgSender() ) revert Scales_YouDontOwnTheseTokens();
```

Comment:
in this case `tokenOwners[tokenId]` will always be "0x000...000", so this would prevent anyone from staking. Further, the `safeTransferFrom` function below ensures a user cannot stake a token they don't own because they can't transfer a token they don't own.